streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -112,6 +112,8 @@
      * These should be valid properties from {@link org.apache.kafka.common.config.TopicConfig TopicConfig}.
      * It is recommended to use {@link #topicPrefix(String)}.
      */
+    // TODO: currently we cannot get the full topic configurations and hence cannot allow topic configs without the prefix,
+    //       this can be lifted once kafka.log.LogConfig is completely deprecated by org.apache.kafka.common.config.TopicConfig
```

Comment:
Don't understand this comment. Can you elaborate?

Reply:
For admin, producer and consumer we have `getXXXConfigs` which relies on `getClientPropsWithPrefix(XXX_PREFIX, XXXConfig.configNames());` However for topic configs we do not have such a class (note `TopicConfig` is not sufficient, and the source of source is still in broker-side `KafkaConfig`), so we cannot have a `getTopicConfigs` that can detect all the topic config names. Hence we ALWAYS have to prefix the config name to indicate it is a topic config name, unlike others which we do not have to.

Reply:
Thx. Got it.